### PR TITLE
Allergens dictionaries translation to italian and reformatting

### PR DESCRIPTION
**Pull request by VaiTon**

### CODE REVIEW

**Comment by VaiTon** [comment]
@stephanegigandet why is it failing?

**Comment by stephanegigandet** [comment]
@VaiTon : I didn't see one change you made, which I think is causing the problem:
"	fr => " de | du | des | d' ",
--> there's an extra space after the ' (same in 4 or 5 other places after d' / l'. Which means it won't match strings like "d'arachide". Can you remove them?

Also, it would be great if you can add a few tests that cover the additions you made in those files:

t/ingredients_parsing.t --> splits the traces
t/allergens_tags.t
t/ingredients_tags.t --> see if it removes the traces from the ingredients_tags

The parsing is quite complex and it's very easy to break something when we add support for something else, so it will be good to have tests.

Thanks!

File: lib/ProductOpener/Ingredients.pm [hunk 1]
```diff
@@ -97,53 +97,53 @@ my $separators = qr/($stops\s|$commas|$separators_except_comma)/i;
...
 my %of = (
-en => " of ",
-de => " von ",
-es => " de ",
-fr => " de | du | des | d'",
-it => " di ",
+	en => " of ",
+	de => " von ",
+	es => " de ",
+	fr => " de | du | des | d' ",
```
Comment (stephanegigandet) [inline lib/ProductOpener/Ingredients.pm]:
```suggestion
	fr => " de | du | des | d'",
```

File: lib/ProductOpener/Ingredients.pm [hunk 2]
```diff
@@ -97,53 +97,53 @@ my $separators = qr/($stops\s|$commas|$separators_except_comma)/i;
...
 my %and_of = (
-en => " and of ",
-de => " und von ",
-es => " y de ",
-fr => " et de | et du | et des | et d'",
-it => " e di ",
+	en => " and of ",
+	de => " und von ",
+	es => " y de ",
+	fr => " et de | et du | et des | et d' ",
```
Comment (stephanegigandet) [inline lib/ProductOpener/Ingredients.pm]:
```suggestion
	fr => " et de | et du | et des | et d'",
```

File: lib/ProductOpener/Ingredients.pm [hunk 3]
```diff
@@ -97,53 +97,53 @@ my $separators = qr/($stops\s|$commas|$separators_except_comma)/i;
...
-en => " of ",
-de => " von ",
-es => " de ",
-fr => " de | du | des | d'",
-it => " di ",
+	en => " of ",
+	de => " von ",
+	es => " de ",
+	fr => " de | du | des | d' ",
+	it => " di | d' ",
```
Comment (stephanegigandet) [inline lib/ProductOpener/Ingredients.pm]:
```suggestion
	it => " di | d'",
```

File: lib/ProductOpener/Ingredients.pm [hunk 4]
```diff
@@ -97,53 +97,53 @@ my $separators = qr/($stops\s|$commas|$separators_except_comma)/i;
...
-en => " and of ",
-de => " und von ",
-es => " y de ",
-fr => " et de | et du | et des | et d'",
-it => " e di ",
+	en => " and of ",
+	de => " und von ",
+	es => " y de ",
+	fr => " et de | et du | et des | et d' ",
+	it => " e di | e d' ",
```
Comment (stephanegigandet) [inline lib/ProductOpener/Ingredients.pm]:
```suggestion
	it => " e di | e d'",
```

File: lib/ProductOpener/Ingredients.pm [hunk 5]
```diff
@@ -97,53 +97,53 @@ my $separators = qr/($stops\s|$commas|$separators_except_comma)/i;
...
 
 
 my %the = (
-en => " the ",
-es => " el | la | los | las ",
-fr => " le | la | les | l'",
-it => " il | la ",
+	en => " the ",
+	es => " el | la | los | las ",
+	fr => " le | la | les | l' ",
```
Comment (stephanegigandet) [inline lib/ProductOpener/Ingredients.pm]:
```suggestion
	fr => " le | la | les | l'",
```

File: lib/ProductOpener/Ingredients.pm [hunk 6]
```diff
@@ -97,53 +97,53 @@ my $separators = qr/($stops\s|$commas|$separators_except_comma)/i;
...
+	it => " il | lo | la | i | gli | le | l' ",
```
Comment (stephanegigandet) [inline lib/ProductOpener/Ingredients.pm]:
```suggestion
	it => " il | lo | la | i | gli | le | l'",
```

**Pull request merged**